### PR TITLE
Adding Gong Oauth Connection 

### PR DESCRIPTION
**Pull request by albandum**
## Description
This PR prepares the terrain for Gong to migrate to the new Oauth API. 
It adds Oauth to Gong transcripts gated behind FF.

There's no need for migrations as for now we have 0 Gong transcripts users - we'll wait until we have a /setup endpoint to keep going on the connection creation.

### CODE REVIEW

File: front/lib/labs/config.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ const config = {
           "NANGO_GOOGLE_DRIVE_CONNECTOR_ID"
         );
       case "gong":
-        return EnvironmentConfig.getEnvVariable("NANGO_GONG_CONNECTOR_ID");
```
Comment (spolu) [inline front/lib/labs/config.ts:18]: You can't really direclty go to a Nango removal. You have to support both flows temporarily until everything is migrated :+1:

File: front/temporal/labs/utils/gong.ts [hunk 2]
```diff
@@ -1,9 +1,44 @@
...
+    });
+    if (tokRes.isErr()) {
+      logger.error(
+        {
+          connectionId: transcriptsConfiguration.connectionId,
+          error: tokRes.error,
+        },
+        "Error retrieving Intercom access token"
+      );
+      throw new Error("Error retrieving Intercom access token");
```
Comment (flvndvd) [inline front/temporal/labs/utils/gong.ts:30]: @albandum you might want to surface this error differently, with the current implementation the activity will always fail... It's okay for this PR but please make sure to find some time to either move this to connectors or implement proper/more robust error handling.

File: front/pages/w/[wId]/assistant/labs/transcripts/index.tsx [hunk 3]
```diff
@@ -347,20 +351,38 @@ export default function LabsTranscriptsIndex({
...
-        );
-        const {
-          connectionId: newConnectionId,
-        }: { providerConfigKey: string; connectionId: string } =
-          await nango.auth(nangoGongConnectorId, nangoConnectionId);
-        await saveOauthConnection(
-          newConnectionId,
-          transcriptsConfigurationState.provider
-        );
+        if (owner.flags.includes("test_oauth_setup")) {
```
Comment (spolu) [inline front/pages/w/[wId]/assistant/labs/transcripts/index.tsx:354]: We can likely just yolo it

**Review by spolu — APPROVED** [review]
LGTM

Provider implementation already done? Or am I missing it?

File: front/pages/w/[wId]/assistant/labs/transcripts/index.tsx (hunk 3 above)
Comment (flvndvd) [reply front/pages/w/[wId]/assistant/labs/transcripts/index.tsx]: Yep updated.

**Comment by flvndvd** [comment]
> LGTM
> 
> Provider implementation already done? Or am I missing it?

Yep done in https://github.com/dust-tt/dust/pull/6413.

**Pull request merged**